src/java.base/share/man/java.md

```diff
@@ -2883,6 +2883,48 @@ Java HotSpot VM.
     expensive operations. Using a lower value will cause heap memory to be
     uncommitted earlier, at the risk of soon having to commit it again.
 
+`-XX:+UseShenandoahGC`
+:   Enables the use of the Shenandoah garbage collector. This is a low pause
+    time, concurrent garbage collector. Its pause times are not proportional to
+    the size of the heap. Shenandoah garbage collector can work with compressed
+    pointers. See `-XX:UseCompressedOops` for further information about
+    compressed oops.
+
+`-XX:ShenandoahGCMode=`*mode*
+:   Sets the GC mode for Shenandoah GC to use. By default, this option is set
+    to `satb`. Among other things, this defines which barriers are in use.
+    Possible mode values include the following:
+
+    `satb`
+    :   Snapshot-at-the-beginning concurrent GC (three pass mark-evac-update).
+
+    `passive`
+    :   Stop the world GC only (either degenerated or full). This mode is
+        diagnostic, and must be enabled via `-XX:+UnlockDiagnosticVMOptions`.
+
+    `generational`
+    :   Generational concurrent GC.
+
+`-XX:ShenandoahGCHeuristics=`*heuristics*
+:   Sets the heuristics for Shenandoah GC to use. By default, this option is
+    set to `adaptive`. This fine-tunes the GC mode selected, by choosing when
+    to start the GC, how much to process on each cycle, and what other features
+    to automatically enable. Possible heuristics are following:
+
+    `adaptive`
+    :   To maintain the given amount of free heap at all times, even during
+    the GC cycle.
+
+    `static`
+    :   Trigger GC when free heap falls below the ShenandoahMinFreeThreshold.
```

Comment:
Same here: `ShenandoahMinFreeThreshold` is experimental, it is not worth mentioning here. In fact, there are tunables for `adaptive` and `compact` as well, yet we do not (correctly!) mention them.

Reply:
True, other options respond to `ShenandoahMinFreeThreshold` as well. It's kinda hard to describe `static` option without mentioning the threshold. 
- We could make the doc vague as in the [shenandoah_globals.hpp](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/gc/shenandoah/shenandoah_globals.hpp#L147): `trigger GC when free heap falls below the threshold`. 
- I now wonder why `static` is a non experimental flag - it has to be used with other experimental flags. Maybe just drop static from man page as well?

Reply:
`below a specified threshold` might sound appropriately vague and render the flag unusable unless one consulted the code, underlining the experimental nature of it.

If the collector respects `static` as specified, it can be considered a good guardrail to use in anger. In fact, I know of teams that do in production. (For that matter, there are a couple who use `compact` as well.)

However, I am fine with eliding all experimental options, and in turn any non-experimental options that depend on experimental options (which however seems to be an anti-pattern to be avoided, if possible).